tests/conftest.py

```diff
@@ -105,3 +122,31 @@ def dummy_private_key():
 K8PZxUBy9cZ0KOEpAkA1b7cZpW40ZowMvAH6sF+7Ok1NFd+08AMXLiSJ6z7Sk29s
 UrfAc2T6ZnfNC4qLIaDyo87CzVG/wk1Upr21z0YD
 -----END RSA PRIVATE KEY-----"""
+
+
+class VersionCheckingAdapter(requests.adapters.HTTPAdapter):
+    def __init__(self, expected_version):
+        self.expected_version = expected_version
+        super().__init__()
+
+    def send(self, request, *args, **kwargs):
+        request.headers["X-Exodus-Query"] = "1"
+        response = super().send(request, *args, **kwargs)
+
+        version = response.headers["X-Exodus-Version"]
+        if self.expected_version not in version:
+            raise AssertionError(
+                "Expected to run against version %s but server sent X-Exodus-Version: %s"
+                % (self.expected_version, version)
+            )
+        return response
+
+
+@pytest.fixture
+def requests_session(src_version):
+    session = requests.Session()
+    if src_version:
```

Comment:
This seems like dead code, because you didn't write src_version to return None if there's no version.  You wrote it to call pytest.skip, so you'd never get here.